src/Functions/csvToJsonString.cpp

```diff
@@ -0,0 +1,307 @@
+        }
+
+        // First argument must be a non-null string (field names)
+        WhichDataType which(arguments[0]);
+        if (!which.isString())
+        {
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "First argument (field names) of function {} must be String constant, got {}",
+                getName(),
+                arguments[0]->getName());
+        }
+
+        // Second argument must be (nullable) string (CSV values)
+        which = WhichDataType(arguments[1]);
+        bool isCSVNullable = which.isNullable();
+        if (!which.isString() && !isCSVNullable)
+        {
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Second argument (csv values) of function {} must be String or Null, got {}",
+                getName(),
+                arguments[1]->getName());
+        }
+
+        // Third argument (if present) must be a boolean (detectTypes)
+        if (arguments.size() == 3)
+        {
+            which = WhichDataType(arguments[2]);
+            if (!which.isUInt8())
+            {
+                throw Exception(
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                    "Third argument (detectTypes) of function {} must be Boolean, got {}",
+                    getName(),
+                    arguments[2]->getName());
+            }
+        }
+
+        // function either returns nullable string if CSV input is nullable, or a string otherwise
+        return isCSVNullable ? makeNullable(std::make_shared<DataTypeString>()) : std::make_shared<DataTypeString>();
+    }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        if (!input_rows_count)
+        {
+            return ColumnString::create(); // fast path for empty input
+        }
+
+        FormatSettings::CSV settings = getFormatSettings(context).csv;
+        std::stringstream dest;
+        bool detectTypes = true;
+
+        // did we receive a non-null option third argument with CSV parsing options?
+        if (arguments.size() == 3)
+        {
+            if (const auto * constOptions = checkAndGetColumnConst<ColumnUInt8>(arguments[2].column.get()); constOptions)
+            {
+                detectTypes = constOptions->getValue<UInt8>() != 0;
+            }
+        }
+
+        // we expect the field names to be a string constant
+        if (const auto * constFieldNames = checkAndGetColumnConst<ColumnString>(arguments[0].column.get()); constFieldNames)
+        {
+            const auto fieldNamesStr = constFieldNames->getDataAt(0);
+            std::vector<std::string> fieldNames;
+            fieldNames.reserve(128);
```

Comment:
Can you shed light on why 128 (vs. 42 or anything else)?  May be worth a preprocessor constant since we have a few places where 128 is used (assuming they're intended to be aligned).